pkg/aws/actuator/actuator_test.go

```diff
@@ -372,6 +377,88 @@ func TestUpgradeable(t *testing.T) {
 		})
 	}
 }
+
+func TestSecretFormat(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+
+	tests := []struct {
+		name            string
+		accessKeyID     string
+		secretAccessKey string
+		existingSecret  *corev1.Secret
+	}{
+		{
+			name:            "new secret with credentials field",
+			accessKeyID:     "AKFIRSTKEY",
+			secretAccessKey: "FIRSTSECRET",
+		},
+		{
+			name:            "existing secret without credentials field",
+			accessKeyID:     "AKFIRSTKEY",
+			secretAccessKey: "FIRSTSECRET",
+			existingSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testTargetSecret,
+					Namespace: testTargetNamespace,
+				},
+				Data: map[string][]byte{
+					"aws_access_key_id":     []byte("SOMEACCESSKEY"),
+					"aws_secret_access_key": []byte("SOMESECRETKEY"),
+				},
+			},
+		},
+		{
+			name:            "existing secret with outdated credentials field",
+			accessKeyID:     "AKFIRSTKEY",
+			secretAccessKey: "FIRSTSECRET",
+			existingSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testTargetSecret,
+					Namespace: testTargetNamespace,
+				},
+				Data: map[string][]byte{
+					"aws_access_key_id":     []byte("SOMEACCESSKEY"),
+					"aws_secret_access_key": []byte("SOMESECRETKEY"),
+					"credentials":           []byte("OLD AWS CONFIG"),
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var fakeClient client.Client
+			if test.existingSecret != nil {
+				fakeClient = fake.NewFakeClient(test.existingSecret)
+			} else {
+				fakeClient = fake.NewFakeClient()
+			}
+
+			a := &AWSActuator{
+				Client: fakeClient,
+			}
+
+			cr := testCredentialsRequest()
+			logger := a.getLogger(cr)
+			err := a.syncAccessKeySecret(cr, test.accessKeyID, test.secretAccessKey, test.existingSecret, "exampleAWSPolicy", logger)
+
+			require.NoError(t, err, "unexpected error creating/updating Secret")
+
+			secret := &corev1.Secret{}
+			secretKey := types.NamespacedName{Name: cr.Spec.SecretRef.Name, Namespace: cr.Spec.SecretRef.Namespace}
```

Comment:
Suggest secretNSN or similar, key might be a little misleading here. I'm thinking Data[key] when I see that var.